test/integration/nested_routes_test.rb

```diff
@@ -0,0 +1,295 @@
+        resources 'variants', only: [:index, :show]
+      end
+      resources :admins, only: [:index, :show], controller: :agents do
+        resources :comments, only: [:index], controller: :topics
+      end
+
+      namespace 'api' do
+        resources 'users', only: [:index] do
+          resources 'comments', only: [:index]
+        end
+      end
+
+      resources :admin do
+        resources :users do
+          collection do
+            get '/search'
+          end
+        end
+      end
+
+      resources :organizations do
+        resources :projects do
+          member do
+            get '/configure'
+          end
+        end
+      end
+
+      namespace :center do
+        resources :tickets, only: [:show], controller: :receipts do
+          resources :customers
+        end
+      end
+    end
+
+    @inspector = @router.inspector.to_s
+    @app = Rack::MockRequest.new(@router)
+  end
+
+  describe 'inspector' do
+    describe 'should print correct routes' do
+      it 'for resources -> resources' do
+        @inspector.must_match 'users_posts GET, HEAD  /users/:users_id/posts         Nested::Controllers::Users::Posts::Index'
+        @inspector.must_match 'new_users_posts GET, HEAD  /users/:users_id/posts/new     Nested::Controllers::Users::Posts::New'
+        @inspector.must_match 'users_posts POST       /users/:users_id/posts         Nested::Controllers::Users::Posts::Create'
+        @inspector.must_match 'users_posts GET, HEAD  /users/:users_id/posts/:id     Nested::Controllers::Users::Posts::Show'
+        @inspector.must_match 'edit_users_posts GET, HEAD  /users/:users_id/posts/:id/edit Nested::Controllers::Users::Posts::Edit'
+        @inspector.must_match 'users_posts PATCH      /users/:users_id/posts/:id     Nested::Controllers::Users::Posts::Update'
+        @inspector.must_match 'users_posts DELETE     /users/:users_id/posts/:id     Nested::Controllers::Users::Posts::Destroy'
+      end
+
+      it 'for resources -> resource' do
+        @inspector.must_match 'new_users_avatar GET, HEAD  /users/:users_id/avatar/new    Nested::Controllers::Users::Avatar::New'
+        @inspector.must_match 'users_avatar POST       /users/:users_id/avatar        Nested::Controllers::Users::Avatar::Create'
+        @inspector.must_match 'users_avatar GET, HEAD  /users/:users_id/avatar        Nested::Controllers::Users::Avatar::Show'
+        @inspector.must_match 'edit_users_avatar GET, HEAD  /users/:users_id/avatar/edit   Nested::Controllers::Users::Avatar::Edit'
+        @inspector.must_match 'users_avatar PATCH      /users/:users_id/avatar        Nested::Controllers::Users::Avatar::Update'
+        @inspector.must_match 'users_avatar DELETE     /users/:users_id/avatar        Nested::Controllers::Users::Avatar::Destroy'
+      end
+
+      it 'for resource -> resources' do
+        @inspector.must_match 'user_comments GET, HEAD  /user/:user_id/comments        Nested::Controllers::User::Comments::Index'
```

Comment:
@AlfonsoUceda Same thing of https://github.com/lotus/router/pull/51/files#r28851650 : `/user/comments`